opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -2141,6 +2141,43 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
     int overlap_cells_per_rank = 0;
     int overlap_plus_neighbors_per_rank = 0;
     (*current_data_)[0]->cellIndexSet().beginResize();
+    auto& leaf_index_set =  (*current_data_).back()->cellIndexSet();
+    leaf_index_set.beginResize();
+    // I am unsure about the order that you are using.
+    // In our simple case we can do
+    
+    for(const auto& element : elements(leafGridView)) {
+        if(!element.hasFather())
+        {
+            leaf_index_set.add(globalIdSet().id(element),
+                               ParallelIndexSet::LocalIndex(element.index(),
+                                                            AttributeSet((element.partitionType()==InteriorEntity)? owner : overlap),
```

Comment:
We might need to get the element on level 0 here to determine the partition type.